src/Product.php

```diff
@@ -214,20 +224,95 @@ public static function display_store_stock_status_block() {
     }
 
     global $product;
+    $store_stocks = static::getStockByStore($product);
 
-    $stocks = static::getStockByStore($product);
-    $raw = $stocks;
-    foreach ($stocks as $name => $types) {
+    $availability = self::buildInitialAvailabilityArray(static::$allVariationsStock);
+    if (!empty(static::$allVariationsStock) && count(static::$allVariationsStock) > 1) {
+      foreach (static::$allVariationsStock as $product_id => $locations) {
+        $availability = self::getAvailabilityWithProductID($availability, $product_id, $locations);
+        $store_stocks = self::getStoreStocksWithUpdatedAvailability($store_stocks, $product_id, $locations);
+      }
+    }
```

Comment:
This appears to prepare additional stock availability data. This should not be done within the display/render function. There are separate functions that are retrieving/generating the data only.

Note that the hook implementation `woocommerce_available_variation` right in this file already prepares the stock by store for each loaded variant. The information is output within the JSON data for all variants in the frontend. Maybe it would be faster and better to process and calculate that existing information in the frontend only?  Otherwise we really need to ensure that this change in the backend does not cause major performance bottleneck (due to loading all variants again).